osu.Game.Tests/Rulesets/TestSceneRulesetSkinProvidingContainer.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+ // See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using JetBrains.Annotations;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.OpenGL.Textures;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Audio;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+using osu.Game.Skinning;
+using osu.Game.Tests.Testing;
+using osu.Game.Tests.Visual;
+
+namespace osu.Game.Tests.Rulesets
+{
+    public class TestSceneRulesetSkinProvidingContainer : OsuTestScene
+    {
+        [Resolved]
+        private SkinManager skins { get; set; }
+
+        private SkinRequester requester;
+
+        protected override Ruleset CreateRuleset() => new TestRuleset();
+
+        [Test]
+        public void TestEarlyAddedSkinRequester()
+        {
+            ISample transformerSampleOnBdl = null;
+
+            // need a legacy skin to plug the TestRuleset's legacy transformer, which is required for testing this.
```

Comment:
Is this actually true? Can you explain why this only broke for legacy transformer cases, not all cases where the `RulesetSkinProvider` was responsible for the lookup?

Or in other words, why couldn't this be tested with a `GetTexture` lookup rather than setting up all the legacy transformers and rulesets. I'm pretty sure it would fail just as well?

Reply:
Good point, got carried away by one way I thought of doing it and didn't realize it could've been done in another easier way.